scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+} else := "enabled but sharing items to non-google accounts is disabled" if {
+    Value_A in AcceptableValues1_4_A
+} else := "enabled and items can be shared to non-google accounts"
+
+NonComplianceMessage1_4(value) := sprintf("External Sharing is %s",
+                                          [value])
+
+NonCompliantOUs1_4 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_4(GetFriendlyValue1_4(LastEvent_A.NewValue,
+                                                         LastEvent_B.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events_A := utils.FilterEventsOU(LogEvents, LogMessage1_4_A, OU)
+    count(Events_A) > 0
+    LastEvent_A := utils.GetLastEvent(Events_A)
+
+    Events_B := utils.FilterEventsOU(LogEvents, LogMessage1_4_B, OU)
+    count(Events_B) > 0
+    LastEvent_B := utils.GetLastEvent(Events_B)
+
+    not LastEvent_A.NewValue in AcceptableValues1_4_A
+    not LastEvent_B.NewValue in AcceptableValues1_4_B
+}
+
+NonCompliantGroups1_4 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage1_4(GetFriendlyValue1_4(LastEvent_A.NewValue,
+                                                         LastEvent_B.NewValue))
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events_A := utils.FilterEventsGroup(LogEvents, LogMessage1_4_A, Group)
+    count(Events_A) > 0
+    LastEvent_A := utils.GetLastEvent(Events_A)
+
+    Events_B := utils.FilterEventsGroup(LogEvents, LogMessage1_4_B, Group)
+    count(Events_B) > 0
+    LastEvent_B := utils.GetLastEvent(Events_B)
+
+    not LastEvent_A.NewValue in AcceptableValues1_4_A
+    not LastEvent_B.NewValue in AcceptableValues1_4_B
+}
+
+NonCompliantOUs1_4 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage1_4(GetFriendlyValue1_4(nonGoogle, ""))
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    ExternalSharingAllowed(OU)
+    nonGoogle := settings.drive_and_docs_external_sharing.allowNonGoogleInvites
+    nonGoogle != false
+}
+
+tests contains {
+    "PolicyId": DriveId1_4,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := false
+    not Check1_4_OK
+}
+
+tests contains {
+    "PolicyId": DriveId1_4,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_4, NonCompliantGroups1_4),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_4,
+        "NonCompliantGroups": NonCompliantGroups1_4},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check1_4_OK
+    Conditions := {count(NonCompliantOUs1_4) == 0, count(NonCompliantGroups1_4) == 0}
+    Status := (false in Conditions) == false
+}
+
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.1.5
+#--
```

Comment:
If you have "When sharing outside...is allowed, users...can make files and published web content visible to anyone with the link" selected, then set outside sharing to OFF, 1.5 still fails.

Reply:
done